repairnator/repairnator-dockerpool/src/main/java/fr/inria/spirals/repairnator/dockerpool/Launcher.java

```diff
@@ -236,15 +237,25 @@ private void printUsage() {
         System.exit(-1);
     }
 
-    private List<Integer> readListOfBuildIds() {
-        List<Integer> result = new ArrayList<>();
+    private List<InputBuildId> readListOfBuildIds() {
+        List<InputBuildId> result = new ArrayList<>();
+
         File inputFile = this.arguments.getFile("input");
 
         try {
             BufferedReader reader = new BufferedReader(new FileReader(inputFile));
             while (reader.ready()) {
                 String line = reader.readLine().trim();
-                result.add(Integer.parseInt(line));
+                String[] buildIds = line.split(",");
+                if (buildIds.length > 0) {
+                    int buggyBuildId = Integer.parseInt(buildIds[0]);
+                    if (launcherMode == LauncherMode.REPAIR) {
+                        result.add(new InputBuildId(buggyBuildId));
+                    } else {
+                        int patchedBuildId = (buildIds.length > 1) ? Integer.parseInt(buildIds[1]) : 0;
```

Comment:
It does not look good: if you're in bears mode you should throw an exception if you cannot reach the patched build id.

Reply:
You are right: and I put a log error message instead of throwing an exception.